src/stream/src/executor/top_n_new.rs

```diff
@@ -88,11 +88,38 @@ pub struct InnerTopNExecutorNew<S: StateStore> {
     /// We are interested in which element is in the range of [offset, offset+limit).
     managed_state: ManagedTopNStateNew<S>,
 
+    /// In-memory cache of top (N + `TOPN_CACHE_DEFAULT_HIGH_CAPACITY`) rows
+    cache: TopNCache,
+
     #[expect(dead_code)]
     /// Indices of the columns on which key distribution depends.
     key_indices: Vec<usize>,
 }
 
+const TOPN_CACHE_DEFAULT_HIGH_CAPACITY: usize = 1024;
```

Comment:
I think a fixed ratio (e.g. N*2) sounds more reasonable than fixed size?

Reply:
Is `TOPN_CACHE_DEFAULT_HIGH_CAPACITY `  related to `offset` and `limit`?

Reply:
> I think a fixed ratio (e.g. N*2) sounds more reasonable than fixed size?

The problem is that I found that if the capacity for `high` is small, for example N*1.2 in design doc, we may need to refill the high cache frequently. I am not sure what coefficient is reasonable here, so prefer a reasonable large fixed value here and hope it can satisfy majority cases.

Reply:
> Is `TOPN_CACHE_DEFAULT_HIGH_CAPACITY ` related to `offset` and `limit`?

The `high` cache is extra memory cache that to eliminate scanning on state store.
In the design doc, the capacity for `high` cache is configured to `(offset + limit) * c` where c is a coefficient.